pkg/api/v1/atlasdeployment_types.go

```diff
@@ -205,9 +214,19 @@ type ServerlessSpec struct {
 	// +kubebuilder:validation:Pattern:=^[a-zA-Z0-9][a-zA-Z0-9-]*$
 	Name string `json:"name"`
 	// Configuration for the provisioned hosts on which MongoDB runs. The available options are specific to the cloud service provider.
-	ProviderSettings *ProviderSettingsSpec `json:"providerSettings"`
-
+	ProviderSettings *ProviderSettingsSpec       `json:"providerSettings"`
 	PrivateEndpoints []ServerlessPrivateEndpoint `json:"privateEndpoints,omitempty"`
+	// Key-value pairs for resource tagging.
+	// +kubebuilder:validation:MaxItems=50
+	// +optional
+	Tags []*TagSpec `json:"tags"`
```

Comment:
should this be `omitempty` too or does serverless behave differently?

Reply:
looking at https://github.com/mongodb/go-client-mongodb-atlas/pull/509, assuming this is intentional